src/mumble/GlobalShortcut_win.cpp

```diff
@@ -18,80 +26,95 @@
 #include <QtCore/QElapsedTimer>
 #include <QtCore/QTimer>
 
-// 3rdparty/xinputcheck-src.
-#include <xinputcheck.h>
+#include <codecvt>
 
-// Used to detect screen readers
-#include <tlhelp32.h>
+extern "C" {
+// Do NOT change the order of the includes below or compile errors will occur!
+#include <hidsdi.h>
 
-#define HAS_FAILED(Status) (static_cast< HRESULT >(Status) < 0)
-
-#define DX_SAMPLE_BUFFER_SIZE 512
+#include <hidpi.h>
+}
 
-// from os_win.cpp
+// From os_win.cpp
 extern HWND mumble_mw_hwnd;
 
 /// The QEvent::Type value to use for InjectKeyboardMessageEvent.
 #define INJECTKEYBOARDMESSAGE_QEVENT (QEvent::User + 200)
-/// The QEvent::Type value to use for InjectKeyboardMouseEvent.
+/// The QEvent::Type value to use for InjectMouseMessageEvent.
 #define INJECTMOUSEMESSAGE_QEVENT (QEvent::User + 201)
+/// The QEvent::Type value to use for InjectHIDMessageEvent.
+#define INJECTHIDMESSAGE_QEVENT (QEvent::User + 202)
 
 /// InjectKeyboardMessageEvent is an event that can be sent to
 /// the GlobalShortcutWin engine to inject a native Windows keyboard
-/// event into GlobalShortcutWin's event stream.
+/// packet into GlobalShortcutWin's event stream.
 class InjectKeyboardMessageEvent : public QEvent {
-	Q_DISABLE_COPY(InjectKeyboardMessageEvent);
+	Q_DISABLE_COPY(InjectKeyboardMessageEvent)
```

Comment:
Why did you remove the semicolon? Seems like an unrelated (and unnecessary) change